.github/workflows/continuous-build.yml

```diff
@@ -20,49 +20,76 @@ jobs:
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2
+
       - name: Setup Go
         uses: actions/setup-go@v2
         with:
           go-version: '^1.15'
+
       - uses: actions/cache@v2
         with:
           path: ~/go/pkg/mod
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
+
       - name: Run tests
         run: go test -cover ./...
+
       - name: Build daemon
         run: go build ./cmd/tracing
-  build_docker:
-    name: Build Docker image
+  build_publish_daemon_image:
+    name: Build and Publish X-Ray daemon docker image (alpha tag) to docker hub and ECR
     needs: build
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2
+
+      - name: Configure AWS Credentials
+        if: ${{ github.event_name == 'push' }}
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to Public ECR
+        if: ${{ github.event_name == 'push' }}
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+
       - name: Login to DockerHub
         if: ${{ github.event_name == 'push' }}
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.DOCKER_USER }}
           password: ${{ secrets.DOCKER_PASS }}
+
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v1
+
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
+
       - name: Cache Docker layers
         uses: actions/cache@v2
         with:
           path: /tmp/.buildx-cache
           key: ${{ runner.os }}-buildx-${{ github.sha }}
           restore-keys: |
             ${{ runner.os }}-buildx-
+
       - name: Build docker image
         uses: docker/build-push-action@v2
         with:
-          push: ${{ github.event_name == 'push' }}
+          context: .
```

Comment:
Think the context and file are the defaults

Reply:
okay will confirm and remove it if they are defaults.